dissect/target/helpers/cyber.py

```diff
@@ -0,0 +1,517 @@
+    "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", "0", "1", "2", "3", "4",
+    "5", "6", "7", "8", "9", "Ç", "ü", "é", "â", "ä", "à", "å", "ç", "ê", "ë", "è", "ï",
+    "î", "ì", "Ä", "Å", "É", "æ", "Æ", "ô", "ö", "ò", "û", "ù", "ÿ", "Ö", "Ü", "¢", "£",
+    "¥", "ƒ", "á", "í", "ó", "ú", "ñ", "Ñ", "ª", "º", "¿", "¬", "½", "¼", "¡", "«", "»",
+    "α", "ß", "Γ", "π", "Σ", "σ", "µ", "τ", "Φ", "Θ", "Ω", "δ", "φ", "ε", "±", "÷", "°",
+    "·", "²", "¶", "⌐", "₧", "░", "▒", "▓", "│", "┤", "╡", "╢", "╖", "╕", "╣", "║", "╗",
+    "╝", "╜", "╛", "┐", "└", "┴", "┬", "├", "─", "┼", "╞", "╟", "╚", "╔", "╩", "╦", "╠",
+    "═", "╬", "╧", "╨", "╤", "╧", "╙", "╘", "╒", "╓", "╫", "╪", "┘", "┌", "█", "▄", "▌",
+    "▐", "▀", "∞", "∩", "≡", "≥", "≤", "⌠", "⌡", "≈", "∙", "√", "ⁿ", "■",
+]
+# fmt: on
+
+NMS_TYPE_EFFECT_SPEED = 4 / 1000
+NMS_JUMBLE_SECONDS = 1
+NMS_JUMBLE_LOOP_SPEED = 35
+NMS_REVEAL_SECONDS = 2
+NMS_REVEAL_LOOP_SPEED = 50
+
+MATRIX_CHARS = list(map(chr, range(0x20, 0x7F)))
+
+MATRIX_MAX_SPEED = 5
+MATRIX_MAX_CASCADES = 600
+MATRIX_MAX_COLS = 20
+MATRIX_FRAME_DELAY = 0.03
+MATRIX_REVEAL_SECONDS = 4
+
+
+class Color(Enum):
+    BLACK = 30
+    RED = 31
+    GREEN = 32
+    YELLOW = 33
+    BLUE = 34
+    MAGENTA = 35
+    CYAN = 36
+    WHITE = 37
+
+
+class CyberIO(StringIO):
+    def __init__(self, color: Optional[Color] = None, mask_space: bool = False, run_at_end: bool = False):
+        self._color = color
+        self._mask_space = mask_space
+        self._run_at_end = run_at_end
+        super().__init__()
+
+    def write(self, s: str) -> int:
+        if self._run_at_end:
+            super().write(s)
+        else:
+            cyber_print(s, self._color, self._mask_space)
+        return len(s)
+
+
+@contextmanager
+def cyber(color: Optional[Color] = Color.YELLOW, mask_space: bool = False, run_at_end: bool = False) -> None:
+    stream = CyberIO(color, mask_space, run_at_end)
+    with redirect_stdout(stream):
+        yield
+
+    if run_at_end:
+        cyber_print(stream.getvalue(), color, mask_space)
+
+
+def cyber_print(buf: str, color: Optional[Color] = None, mask_space: bool = False) -> None:
+    if not buf or buf == "\n":
+        sys.__stdout__.write(buf)
+        return
+
+    if not CAN_CYBER:
+        sys.__stdout__.write("you're not cybering hard enough\n")
+
+    if os.getenv("CYBER") == "💊":
+        matrix(buf, color, mask_space)
+    else:
+        nms(buf, color, mask_space)
+
+
+# https://github.com/bartobri/libnms
+def nms(buf: str, color: Optional[Color] = None, mask_space: bool = False) -> None:
+    orig_row, orig_col = (0, 0)
+    with _set_terminal():
+        max_rows, max_cols = _get_win_size()
+
+        orig_row, _ = _get_cursor_pos()
+
+        _cursor_hide()
+        _cursor_move(orig_row, orig_col)
+
+        characters, remaining, (orig_row, orig_col), _ = _get_character_info(
+            buf, max_rows, max_cols, orig_row, orig_col
+        )
```

Comment:
There's a combination of setting stuff up and retrieving some info that's used slightly differently. I'm fine with this small amount of repetition because it's also pretty concise.